ompi/mca/coll/han/coll_han_scatterv.c

```diff
@@ -144,8 +143,6 @@ int mca_coll_han_scatterv_intra(const void *sbuf, const int *scounts, const int
             low_scounts[low_peer] = scounts[w_peer];
         }
 
-        ompi_datatype_type_size(sdtype, &sdsize);
```

Comment:
Same comment as for gather except that you should use unpack to go from a packed buffer into the local type.

Reply:
Currently inside ompi we need extra data size check in order to use MPI_PACKED - it is currently possible that the total byte size of `datatype size` x `count` can exceeded INT_MAX.

I think we should switch to explicit pack/unpack after large count support.

Reply:
@bosilca I took time to come up with possible optimizations following our discussion on Monday. I was trying to take advantage of the type map, but soon realized that this is not generally useful between node leader and follower.

For scatterv, the invariant is actually between Root and other processes:
> The type signature implied by sendcount[i], sendtype at the root must be equal to the type signature implied by recvcount, recvtype at MPI process i (however, the type maps may be different). This implies that the amount of data sent must be equal to the amount of data received, pairwise between each MPI process and the root. Distinct type maps between sender and receiver are still allowed.

When we focus on the node leader and its local neighbors, this information is not helpful, since they do not need to match in their respective `recvcount, recvtype`.

There is a simplification(rather than optimization) opportunity though, if and only if the node leader's `recvtype`'s type map size is the same than its local followers. In that case we don't need `MPI_BYTE` - we can safely use `recvtype` instead. This wouldn't bring better performance though.

Reply:
Because the typemap provided by the different processes must match when complemented with the count, we could use `MPI_Get_elements` to build any datatype used in the communication for as long as we know the size in bytes we are supposed to handle.